### PR TITLE
[Storage] Switch away from pingcap RocksDB

### PR DESCRIPTION
**Pull request by wqfish**
This change switches the RocksDB binding from the current pingcap one to https://github.com/rust-rocksdb/rust-rocksdb.

Note that it uses my personal fork for now, but the branch has only a single commit that will be merged soon.

### CODE REVIEW

File: storage/schemadb/src/lib.rs [hunk 1]
```diff
@@ -204,26 +185,15 @@ impl DB {
...
-        }
-
-        // If db doesn't exist, create a db first with all column families.
-        db_opts.create_if_missing(true);
-
-        let mut db = DB::open_cf(db_opts, path, vec![DEFAULT_CF_NAME])?;
-        column_families
-            .into_iter()
-            .filter(|cf_name| *cf_name != DEFAULT_CF_NAME)
-            .map(|cf_name| db.create_cf(cf_name))
```
Comment (msmouse) [inline storage/schemadb/src/lib.rs:225]: Curious why did we need to create_cf explicitly before?

Comment (wqfish) [reply storage/schemadb/src/lib.rs:225]: The library we were using does not have the `create_missing_column_families` option. Let me double check how this one is implemented.

Comment (wqfish) [reply storage/schemadb/src/lib.rs:225]: https://github.com/facebook/rocksdb/blob/243852ec15f96de760df8612b6ae9f4664297e55/include/rocksdb/options.h#L385

**Comment by wqfish** [comment]
cc @mimoo

**Comment by mimoo** [comment]
Do we still have to do libc++ shenanigans with the new bindings? 

https://github.com/google/oss-fuzz/blob/master/projects/libra/build.sh

in it we:

* had to patch rust-rocksdb to avoid linking libc++ statically
* added `export RUSTFLAGS="-C link-arg=-L/usr/local/lib -C link-arg=-lc++"` at the end to make sure libc++ was linked

**Pull request closed without merging**